src/basic/io-util.c

```diff
@@ -105,18 +106,20 @@ int loop_read_exact(int fd, void *buf, size_t nbytes, bool do_poll) {
         return 0;
 }
 
-int loop_write(int fd, const void *buf, size_t nbytes, bool do_poll) {
+int loop_write_full(int fd, const void *buf, size_t nbytes, bool do_poll, usec_t timeout) {
```

Comment:
somehow github ate my earlier comment: we should remove do_poll entirely, timeout can take it's place. timeout=0 would be as do_poll=false was before, and timeout=USEC_INFINITY as do_poll=true.